app/services/hotkeys.ts

```diff
@@ -240,166 +284,179 @@ export class HotkeysService extends StatefulService<IHotkeysServiceState> {
     this.sourcesService.sourceRemoved.subscribe(() => this.invalidate());
   }
 
-
   addHotkey(hotkeyModel: IHotkey) {
     this.ADD_HOTKEY(hotkeyModel);
   }
 
-
   private invalidate() {
     this.registeredHotkeys = null;
   }
 
   private updateRegisteredHotkeys() {
     const hotkeys: IHotkey[] = [];
+    /*
+     * Since we're hybrid at this point, track already-added hotkeys so OBS
+     * hotkeys don't duplicate them
+     */
+    const addedHotkeys = new Set<string>();
 
-    HOTKEY_ACTIONS.GENERAL.forEach(action => {
+    Object.values(GENERAL_ACTIONS).forEach(action => {
```

Comment:
AFAIK this is the only place you're using the `*_ACTIONS` objects so i'm kind of wondering why you refactored them into objects? if you only use them to turn back into arrays it just seems like it's adding in an extra step, but i might be missing something here

Reply:
We will peel them off, one by one. Easier to have them separate to differentiate logic (such as general being run first and not having to duplicate-check), does that make sense?

Reply:
i understand breaking them up into chunks, but the data transformation from `array` into `object -> array` is where i'm lost

Reply:
We need these indexed for `getActionFromName` and normalization, in the future, this could be a way to improve performance in lookups as well.